endpoints/openrtb2/auction_test.go

```diff
@@ -123,45 +99,49 @@ func TestJsonSampleRequests(t *testing.T) {
 			"currency-conversion/custom-rates/errors",
 		},
 	}
+
+	paramsValidator := newParamsValidator(t)
+
 	for _, test := range testSuites {
 		testCaseFiles, err := getTestFiles(filepath.Join("sample-requests", test.sampleRequestsSubDir))
 		if assert.NoError(t, err, "Test case %s. Error reading files from directory %s \n", test.description, test.sampleRequestsSubDir) {
-			for _, file := range testCaseFiles {
-				data, err := ioutil.ReadFile(file)
-				if assert.NoError(t, err, "Test case %s. Error reading file %s \n", test.description, file) {
-					runTestCase(t, data, file)
-				}
-			}
-		}
-	}
-}
+			for _, testFile := range testCaseFiles {
+				fileData, err := ioutil.ReadFile(testFile)
+				if assert.NoError(t, err, "Test case %s. Error reading file %s \n", test.description, testFile) {
+					// Retrieve values from JSON file
+					test, err := parseTestFile(fileData, testFile)
+					if !assert.NoError(t, err) {
+						continue
+					}
 
-func getTestFiles(dir string) ([]string, error) {
-	var filesToAssert []string
+					auctionEndpointHandler, appNexusServer, openXServer, rubiconServer, mockCurrencyRatesServer, err := buildTestEndpoint(test, paramsValidator)
+					if !assert.NoError(t, err) {
+						continue
+					}
 
-	fileList, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, err
-	}
+					runTestCase(t, auctionEndpointHandler, test, fileData, testFile)
 
-	// Append the path of every file found in `dir` to the `filesToAssert` array
-	for _, fileInfo := range fileList {
-		filesToAssert = append(filesToAssert, filepath.Join(dir, fileInfo.Name()))
+					appNexusServer.Close()
+					openXServer.Close()
+					rubiconServer.Close()
+					mockCurrencyRatesServer.Close()
```

Comment:
Do you think these closing functions can go inside `buildTestEndpoint` func like it was before `defer appNexusServer.Close()`? You will not need to return all servers you need to close, there are no other usages of these variables.

Reply:
From [this article](https://www.geeksforgeeks.org/defer-keyword-in-golang/): "defer statements delay the execution of the function or method or an anonymous method until the nearby function returns". Therefore, I assume a `defer` statement inside `buildTestEndpoint(test testCase, paramValidator openrtb_ext.BidderParamValidator)` prevents a given server from listening as soon as we exit said function. But I agree with you that the `Close()` statements were misplaced because an error thrown by `buildTestEndpoint(test testCase, paramValidator openrtb_ext.BidderParamValidator)` would prevent us from closing them altogether. I corrected accordingly. Please let me know if you agree with the new approach.

Reply:
> Edit: and for auction_benchmark_test.go too.
I agree. Given that from `BenchmarkValidWholeExemplary` we exit with `b.Fatal(err.Error())` in case `buildTestEndpoint` returns an error, the `defer` keyword fits our needs just fine. Corrected.

Reply:
Right, defer will be executed at the closing bracket of the function, but you need to have those servers(not variables) available after buildTestEndpoint. I see, you are right, thank you for checking this!

Reply:
Will approve it if there will be no changes required from other reviewers.